CMakeLists.txt

```diff
@@ -35,24 +39,47 @@ find_package(Threads REQUIRED)
 # SDL2
 find_package(SDL2 REQUIRED)
 
-# OpenAL
+# OpenAL
 find_package(OpenAL REQUIRED)
-add_library(OpenAL::OpenAL UNKNOWN IMPORTED)
+
+# Mac OS frameworks should be handled in a special way. 
+# See more info about cmake issues with Mac OS frameworks here:
+# https://cmake.org/pipermail/cmake/2016-April/063179.html
+# and here
+# https://gitlab.kitware.com/cmake/cmake/-/issues/19410
+if(MAC_OS AND OPENAL_LIBRARY MATCHES "\\.framework$")
+  add_library(OpenAL::OpenAL INTERFACE IMPORTED)
+  set_target_properties(OpenAL::OpenAL PROPERTIES
+    INTERFACE_LINK_LIBRARIES "${OPENAL_LIBRARY}")
+else()
+  add_library(OpenAL::OpenAL UNKNOWN IMPORTED)
+  set_target_properties(OpenAL::OpenAL PROPERTIES
+    IMPORTED_LOCATION ${OPENAL_LIBRARY})
+endif()
+
 set_target_properties(OpenAL::OpenAL PROPERTIES
-  IMPORTED_LOCATION ${OPENAL_LIBRARY}
   INTERFACE_INCLUDE_DIRECTORIES ${OPENAL_INCLUDE_DIR})
 
 # FFmpeg
 find_package(FFmpeg REQUIRED COMPONENTS
   AVCODEC AVFORMAT AVUTIL SWSCALE)
 
-
 # swresample
 find_package(Libswresample REQUIRED)
 add_library(SWRESAMPLE::SWRESAMPLE UNKNOWN IMPORTED)
 set_target_properties(SWRESAMPLE::SWRESAMPLE PROPERTIES
   IMPORTED_LOCATION ${LIBSWRESAMPLE_LIBRARIES}
   INTERFACE_INCLUDE_DIRECTORIES ${LIBSWRESAMPLE_INCLUDE_DIRS})
+
+# MacOSUtils
+if(MAC_OS)
+  add_library(MacOSUtils macOS/src/PathUtils.m)
```

Comment:
The directory is called `macos` in lowercase, so this line breaks the build on Linux